OpenTap.Metrics/AssetDiscovery/AssetDiscoveryManager.cs

```diff
@@ -0,0 +1,54 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace OpenTap.Metrics.AssetDiscovery;
+
+public static class AssetDiscoveryManager
+{
+    private static TraceSource log = OpenTap.Log.CreateSource("AssetDiscovery");
+
+    /// <summary>
+    /// Returns all discovered assets from all available providers.
+    /// </summary>
+    public static Dictionary<IAssetDiscovery, DiscoveryResult> DiscoverAllAssets()
+    {
+        Dictionary<IAssetDiscovery, DiscoveryResult> assets = new Dictionary<IAssetDiscovery, DiscoveryResult>();
+        foreach (var provider in GetAssetDiscoveryProviders())
+        {
+            try
+            {
+                var result = provider.DiscoverAssets();
+                assets[provider] = result;
+            }
+            catch (Exception ex)
+            {
+                log.Error($"Error while discovering assets from {provider.GetType().Name}: {ex.Message}");
+                assets[provider] = new DiscoveryResult
+                {
+                    IsSuccess = false,
+                    Error = ex.GetType().Name
```

Comment:
@rmadsen-ks what does ToString turn into? I was going for just the exception type name, and not the exeption message, since this is going to show up in the Status column here:
![image](https://github.com/user-attachments/assets/3a8d9bed-7f53-4e55-9858-34cf3b3d9640)

Reply:
Is it? This is a list of resources, the discovery result is once per asset provider. 

If the field is called Error it should probably not just be e.g "Exception", but the actual error message. I guess it might be ex.Message then.

Reply:
Yes it is. We expect all asset providers to return the same ID for the same asset, then we use Priority to determine which one gets shown in the list.

I just wanted it to be a very short message, but ok. We can also put the exception message, that makes it easier for the plugin to control.